### PR TITLE
Bug 1832324 - Add a way to await ping submission in iOS 

### PR DESCRIPTION
**Pull request by travis79**
This simply awaits the internal dispatcher which is where the pings are submitted from.

### CODE REVIEW

**Comment by badboy** [comment]
While this might get the job done for the particular feature request, I wonder if this also exposes too much internals so to speak?
We already have `Glean.shutdown()` on the RLB, which does a bunch more, but IMO is the more appropriate API to expose.
Also the Swift dispatcher is only used for the uploader anyway, so this really just exposes waiting for upload anyway.
The shutdown API is more robust, including timeouts, etc.

what do you think?

**Comment by travis79** [comment]
> While this might get the job done for the particular feature request, I wonder if this also exposes too much internals so to speak? We already have `Glean.shutdown()` on the RLB, which does a bunch more, but IMO is the more appropriate API to expose. Also the Swift dispatcher is only used for the uploader anyway, so this really just exposes waiting for upload anyway. The shutdown API is more robust, including timeouts, etc.
> 
> what do you think?

I considered that and have no issues with exposing it through the Swift API, and I agree, it is a more robust solution. The direction I went in this PR currently was only an attempt at the simplest possible solution. 

I'll rework this exposing `shutdown()` and see how it looks.

File: glean-core/ios/GleanTests/GleanTests.swift [hunk 1]
```diff
@@ -367,4 +368,63 @@ class GleanTests: XCTestCase {
         Glean.shared.initialize(uploadEnabled: true, buildInfo: stubBuildInfo())
         XCTAssertFalse(Glean.shared.isCustomDataPath)
     }
+
+    func testShutdown() {
+        // This test relies on Glean not being initialized
+        Glean.shared.testDestroyGleanHandle()
+
+        // We expect only a single ping later
```
Comment (badboy) [inline glean-core/ios/GleanTests/GleanTests.swift:376]: Weeeell, we ignore a baseline ping, so it's more like 2 pings.

**Review by badboy — CHANGES_REQUESTED** [review]
Changelog entry please.

**Review by badboy — APPROVED** [review]

**Pull request merged**